src/transformers/modeling_utils.py

```diff
@@ -852,6 +851,13 @@ def save_pretrained(
                 need to replace :obj:`torch.save` by another method.
             push_to_hub (:obj:`bool`, `optional`, defaults to :obj:`False`):
                 Whether or not to push your model to the Hugging Face model hub after saving it.
+
+                .. warning::
+
+                    Using :obj:`push_to_hub=True` might be unsafe if the folder you are passing along is not in sync
+                    with the repository you want to push to (it will pull the repo before saving to be able to push
+                    after the model is saved, so it might erase some uncommitted changes in that folder)
```

Comment:
hmm this shouldn't be the case, are you sure?

Reply:
Can't reproduce how I arrived to this so let's say it's not ;-)